apps/src/lib/levelbuilder/lesson-editor/ActivitySectionCardButtons.jsx

```diff
@@ -0,0 +1,162 @@
+    this.setState({
+      tipToEdit: {
+        key: `tip-${this.generateTipKey()}`,
+        type: 'teachingTip',
+        markdown: ''
+      },
+      addTipOpen: true
+    });
+  };
+
+  generateTipKey = () => {
+    let tipNumber = this.props.activitySection.tips.length + 1;
+    while (
+      this.props.activitySection.tips.some(
+        tip => tip.key === `tip-${tipNumber}`
+      )
+    ) {
+      tipNumber++;
+    }
+
+    return tipNumber;
+  };
+
+  handleCloseAddTip = tip => {
+    if (!this.state.editingExistingTip) {
+      this.props.addTip(tip);
+    }
+    this.setState({addTipOpen: false, editingExistingTip: false});
+  };
+
+  handleOpenAddResource = () => {
+    this.setState({addResourceOpen: true});
+  };
+
+  handleCloseAddResource = () => {
+    this.setState({addResourceOpen: false});
+  };
+
+  render() {
+    return (
+      <div>
+        <div style={styles.bottomControls}>
+          <span>
+            <button
+              onMouseDown={this.handleOpenAddLevel}
+              className="btn"
+              style={styles.addLevel}
+              type="button"
+            >
+              <i style={{marginRight: 7}} className="fa fa-plus-circle" />
+              Add Level
+            </button>
+            <button
+              onMouseDown={this.handleOpenAddTip}
+              className="btn"
+              style={styles.addLevel}
+              type="button"
+            >
+              <i style={{marginRight: 7}} className="fa fa-plus-circle" />
+              Add Tip
+            </button>
+            <button
+              onMouseDown={this.handleOpenAddResource}
+              className="btn"
+              style={styles.addLevel}
+              type="button"
+            >
+              <i style={{marginRight: 7}} className="fa fa-plus-circle" />
+              Add Resource Link
+            </button>
+          </span>
+          {this.props.activitySection.tips.length > 0 && (
+            <span>
+              {this.props.activitySection.tips.map(tip => {
+                return (
+                  <TipWithTooltip
+                    tip={tip}
+                    key={tip.key}
+                    onClick={this.handleEditTip}
+                  />
+                );
+              })}
+            </span>
+          )}
+        </div>
+        <AddResourceDialog
+          isOpen={this.state.addResourceOpen}
+          handleConfirm={this.handleCloseAddResource}
+        />
+        <EditTipDialog
+          isOpen={this.state.addTipOpen}
+          handleConfirm={this.handleCloseAddTip}
+          tip={this.state.tipToEdit}
+        />
+        <AddLevelDialog
+          isOpen={this.state.addLevelOpen}
+          handleConfirm={this.handleCloseAddLevel}
+          currentLevels={this.props.activitySection.levels}
+          addLevel={this.props.addLevel}
+          activitySectionPosition={this.props.activitySection.position}
+          activityPosition={this.props.activityPosition}
```

Comment:
sorry to ask this, but can you please split the AddLevelDialog and everything within it into a separate PR? That would include the following components and their tests:
* AddLevelDialog
* AddLevelFilters
* AddLevelTable
* CreateNewLevelInputs

in case you want more specific guidance while folks may not be online over the weekend, I would suggest something like this sequence (apologies if it is too much detail):
* `git checkout -b create-activity-editor-bak` (create a backup)
* `git checkout -` (back to this branch)
* delete AddLevelDialog, etc
* `git commit -A`
* `git status` (make sure no files were left behind)
* `git checkout -b add-level-dialog`
* `git revert HEAD`
* make sure everything looks good
* `git branch -D create-activity-editor-bak`

now you should have all the AddLevelDialog stuff in a separate branch, ready to open a PR against with this branch as the base.

Reply:
Done!